spec/double_entry/reporting_spec.rb

```diff
@@ -201,6 +201,34 @@
         end
       end
 
+      context 'filtering by key with multiple values' do
+        let(:code) { :spend }
+        let(:filter) do
+          [
+            :metadata => {
+              :category => category,
```

Comment:
It will be a useful feature, but as you mentioned, there is the decision around matching all categories or any category.
Another PR welcome! :trollface: